components/TodoListItem.js

```diff
@@ -1,55 +1,75 @@
-/* eslint-disable @next/next/no-img-element */
+import axios from '../utils/axios'
+import { useState } from 'react'
+import { useAuth } from '../context/auth'
+import {toast} from 'react-toastify'
+import 'react-toastify/dist/ReactToastify.css'
 
-export default function TodoListItem() {
+toast.configure()
+export default function TodoListItem(props) {
+  const {config,API_BASE_URL  } = useAuth()
+  const [Newdata, setNewdata] = useState(props.todo.title);
+  const [editnow, seteditnow] = useState(false);
   const editTask = (id) => {
-    /**
-     * @todo Complete this function.
-     * @todo 1. Update the dom accordingly
-     */
+    seteditnow(true);
   }
 
   const deleteTask = (id) => {
-    /**
-     * @todo Complete this function.
-     * @todo 1. Send the request to delete the task to the backend server.
-     * @todo 2. Remove the task from the dom.
-     */
+    axios
+    .delete(API_BASE_URL + "todo/" + id + "/",config)
+    .then(function ({ data, status }) {
+        toast.success('Task deleted!',{position: toast.POSITION.BOTTOM_RIGHT})
+        props.onDelete(props.todo);
+    })
+    .catch(function (err) {
+      toast.error("An error occurred!");
+    });
   }
 
   const updateTask = (id) => {
-    /**
-     * @todo Complete this function.
-     * @todo 1. Send the request to update the task to the backend server.
-     * @todo 2. Update the task in the dom.
-     */
+    if (!Newdata) {
+        return;
```

Comment:
Warning toast missing!

Reply:
Added warning toast in update method. Please, review again.